charts/datadog/templates/cluster-agent-rbac.yaml

```diff
@@ -115,6 +115,17 @@ rules:
   - "get"
   - "watch"
 {{- end }}
+{{- if and .Values.datadog.apm.instrumentation.enabled .Values.datadog.apm.instrumentation.language_detection.enabled }}
+- apiGroups:
+  - "apps"
+  resources:
+  - deployments
+  verbs:
+  - list
+  - get
+  - watch
+  - patch
```

Comment:
should we spit it in 2? 

I think the current issue is that by default the cluster-agent try to watch the deployment because DD_LANGUAGE_DETECTION_REPORT_ENABLED is true by default.

Reply:
I resolved the issue by setting DD_LANGUAGE_DETECTION_REPORT_ENABLED to false in _common_env if not both apm SSI and language detection is enabled.

In this way we will be always setting it to false everywhere when it shouldn't be set to true.